src/Vision/grippyCode.py

```diff
@@ -0,0 +1,168 @@
+#DONEEEEEEEE!!!!!!!!
+#   (find center of each pair then identify which pair is closest to image center)
+def findPairs(contourList):
+    rightRectList = []
+    leftRectList = []
+    rectPairList = []
+
+    #print ("Number of Rectangles: {}".format(len(contourList)))
+
+    for contour in contourList:
+        rectangleBox = cv2.minAreaRect(contour)
+ 
+        if rectangleBox[2] > rightAngle - deadzone and rectangleBox[2] < rightAngle + deadzone:
+            rightRectList.append(rectangleBox)
+       
+        if rectangleBox[2] > leftAngle - deadzone and rectangleBox[2] < leftAngle + deadzone:
+            leftRectList.append(rectangleBox) 
+
+    for leftRect in leftRectList:
+        pair = (leftRect,None)
+        for rightRect in rightRectList:
+            #0 is center and 0 is x coordinate
+            if rightRect[0][0] > leftRect[0][0]:
+                if pair[1] is None or rightRect[0][0] < pair[1][0][0]:
+                    pair = (leftRect,rightRect)
+        if pair[1] != None:
+            rectPairList.append(pair)
+
+    closestCenterPair = None
+    for pair in rectPairList:
+        print ("rectangle pair coordinates: {}, {}".format(pair[0][0][0],pair[1][0][0]))       
+        if closestCenterPair is None or findPairOffset(pair) < findPairOffset(closestCenterPair):
+            closestCenterPair = pair
+            return closestCenterPair
+
+
+#avgPoint = pairmidpoint
+#coordinate i have then subtract center in pixels to find midpoint
+
+def normalizeImage(pairMidpoint):
+    width = frame.shape[1]
+    imageMidpoint = (width / 2)
+    pairPointOrient = (pairMidpoint - imageMidpoint)
+    fullPairMidpoint = (pairPointOrient / imageMidpoint)
+    # prints after ewww math THIS IS THE CONTOUR MIDPOINT from...
+    # Contour Midpoint Scale (-1 to 1)
+    return fullPairMidpoint
+# MAIN LOOP
+while(True):
+    #capture image
+    ret, frame = cap.read()
+    cv2.waitKey(300)
+
+    #process with GRIP stuff
+    goalFinder.process(frame)
+
+    #filter returned contours
+    filteredContours=[]
+    for contour in goalFinder.find_contours_output:
+        area=cv2.contourArea(contour)
+        if area>1000:
+            filteredContours.append(contour)
+
+    #find best pair, if we found contours
+    if len(filteredContours) >= 2:
+        bestPair = findPairs(filteredContours)
+        if bestPair != None:
+            rectangleMidpoint = (int((bestPair[1][0][0] + bestPair[0][0][0]) / 2), int((bestPair[1][0][1] + bestPair[0][0][1]) / 2))
+            table.putNumber("Vision Correction", normalizeImage(rectangleMidpoint[0]))
+            cv2.circle(frame, (rectangleMidpoint), 7, (0, 255, 0), -1)
+
+            #should put a value from -1 to 1 depending on pair midpoint offset from image midpoint
+    
+
+    #print('Final Pair Midpoint: {},{} ', fullPairMidpoint)
+
+    #show image!
+    cv2.imshow('image',frame)
+
+
+
+cv2.waitKey(0)
+cap.release()
+cv2.destroyAllWindows()
+
+
+"""
```

Comment:
Delete this code

Reply:
deleted line 129

Reply:
I meant to delete the stuff that was all commented out...

Reply:
Should probably re-add that line, and delete everything within `"""`

Reply:
Fixed
😼